sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -699,12 +699,22 @@ public void flush(long timeoutMillis) {
     boolean samplingDecision = tracingSampler.sample(samplingContext);
     transactionContexts.setSampled(samplingDecision);
 
-    SentryTransaction transaction = new SentryTransaction(transactionContexts, this);
-    final StackItem item = stack.peek();
-    if (item != null) {
-      item.scope.setTransaction(transaction);
+    ITransaction transaction = null;
+    if (!isEnabled()) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.WARNING,
+              "Instance is disabled and this 'startTransaction' returns a no-op.");
+      transaction = new NoOpTransaction();
     } else {
-      options.getLogger().log(SentryLevel.FATAL, "Stack peek was null when startTransaction");
+      final StackItem item = stack.peek();
+      if (item != null) {
+        transaction = new SentryTransaction(transactionContexts, this);
+        item.scope.setTransaction(transaction);
+      } else {
+        options.getLogger().log(SentryLevel.FATAL, "Stack peek was null when startTransaction");
```

Comment:
should we do `transaction = new NoOpTransaction();` here too? otherwise it could return null

Reply:
I left it intentionally untouched as merging https://github.com/getsentry/sentry-java/pull/1127 removes this use case.